lib/types/api.d.ts

```diff
@@ -0,0 +1,41 @@
+interface IOption<T> {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Lets make the configuration options explicit instead of Generic